dataset/jeopardy_dataset.py

```diff
@@ -0,0 +1,203 @@
+        word_freq = Counter([])
+        for i, ann in tqdm(enumerate(self.answers)):
+            for answer in ann['answers']:
+                actual_ans = answer['answer']
+                if only_one_word_answers and " " in actual_ans:
+                        continue
+                if only_yes_confidence:
+                    if answer['answer_confidence'] != 'yes':
+                        continue
+                word_freq[actual_ans] += 1
+    
+        return set([word for word in word_freq if word_freq[word] >= threshold])
+        
+
+    def _find_answers(self):
+        """
+            There several answers for each question in the dataset, for example, 
+            here's entry 0 of the answer vector for a random question:
+            {"answer": "net", "answer_confidence": "maybe", "answer_id": 1}
+            
+            For the sake of simplicity, we only consider answers that have answer_confidence: yes,
+            and are exactly one word long. We will apply a similar padding paradigm as we do with the question
+            texts later on, by using the _pad_arr function. One to one correspondence between answers and questions, 
+            as in there is one answers object for each question, and so its fair to go only until the test split for both.
+        """
+        
+        question_to_answer = {}
+        num_unknowns = 0
+        for i, ann in tqdm(enumerate(self.answers)):
+            question_to_answer[ann["question_id"]] = UNK_TOKEN
+            for answer in ann['answers']:
+                actual_ans = answer['answer']
+                if actual_ans in self.frequent_answers:
+                    question_to_answer[ann["question_id"]] = actual_ans
+                    break
+            if question_to_answer[ann["question_id"]] == UNK_TOKEN:
+                num_unknowns += 1
+        print("{} unknown answers, {} known answers".format(str(num_unknowns), str(len(self.answers) - num_unknowns)))
+        return question_to_answer
+
+    def _find_images(self):
+        id_to_filename = {}
+        for filename in os.listdir(self.images_dir):
+            if not filename.endswith('.jpg'):
+                continue
+            id_and_extension = filename.split('_')[-1]
+            image_id = int(id_and_extension.split('.')[0])
+            id_to_filename[image_id] = filename
+        return id_to_filename
+    
+    def __len__(self):
+        return self.i
+    
+    def _process_question(self, question_text):
+        tokenized_question = self.tokenizer(
+            question_text, return_tensors='pt', padding='max_length', truncation=True, max_length=self.q_len+2)
+        contrastive_input = tokenized_question["input_ids"].clone()
+        collator = DataCollatorForLanguageModeling(
+            self.tokenizer,
+            mlm=True,
+            mlm_probability=0.15,
+        )
+        input_ids, labels = collator.mask_tokens(contrastive_input)
```

Comment:
how does this interact with attention masks? Will `contrastive_input` and `input_ids` both have the same masks?

Reply:
I don't recall what we talked about wrt this - the answer to your question is yes though - same attention mask for MLM and contrastive loss. Is this not what I was supposed to do?